add method `getSubscriberTags` to list the tags for a subscriber

ericcarraway opened the pull request:
#### DESCRIPTION:

The existing method `getSubscriber` makes a GET request to `/subscribers/<subscriber_id>` and returns data for a single subscriber.  This endpoint does not, however, return the details of a subscriber's tags.

This PR adds the method `getSubscriberTags`, which makes a GET request to `/subscribers/<subscriber_id>/tags`.  The ConvertKit API responds with an array of tag objects, each including
- a tag `id` (number)
- a tag `name` (string)
- a tag `created_at` (ISO date string)

https://developers.convertkit.com/#list-tags-for-a-subscriber

![PixelSnap 2021-03-28 at 22 00 29@2x](https://user-images.githubusercontent.com/6632454/112766562-a8c18d00-9012-11eb-9499-228f5d56c739.png)

ericcarraway commented on .env.example:
```diff
@@ -0,0 +1,7 @@
+CONVERTKIT_TESTING_EMAIL=
+CONVERTKIT_TESTING_FORM_ID=
+CONVERTKIT_TESTING_SUBSCRIBER_ID=
+CONVERTKIT_TESTING_TAG_ID=
```
It may be helpful to add this `.env.example` file so that future users who wish to run the tests in `index.integration.ts` know how to set up their `.env` file.

jrbaudin replied:
Good point ✅ agreed!

ericcarraway commented on src/models/subscriber/index.ts:
```diff
@@ -42,6 +42,27 @@ export const getSubscriber = async ({ context, input }: types.GetSubscriber): Pr
+export const getSubscriberTags = async ({
+  context,
+  input,
+}: types.GetSubscriber): Promise<types.GetSubscriberTagsResponse> => {
```
RE: `types.GetSubscriber`

The `input` for `getSubscriberTags` is the same as `getSubscriber`.  Essentially, all it needs is a ConvertKit subscriber ID.

Should there be a separate type, or should `types.GetSubscriber` also be used here?

jrbaudin replied:
I personally prefer to have separate types if they diverge in the future. Also it is, to me, more readable.

ericcarraway replied:
Good points.  Addressed in commit 0dee843b89a80c3bde037b47c6ecd367e57458d8. 👍

ericcarraway commented on src/models/subscriber/types.ts:
```diff
@@ -2,6 +2,12 @@ import * as rootTypes from '../../types'
+  name: string
+  created_at: string
+}
+
```
Here's the shape of the response from ConvertKit.  Is `string` the best type of `created_at`, or is there a type that would be better suited for a JSON date string?

![PixelSnap 2021-03-28 at 22 24 37@2x](https://user-images.githubusercontent.com/6632454/112766995-a06a5180-9014-11eb-9fd7-dfb64633c4a4.png)

jrbaudin replied:
In plain typescript I haven't found a better option than `string`. I usually use [`io-ts`](https://gcanti.github.io/io-ts/) though where a runtime ISO date (for example) can be utilized. That's besides the point though. `string` is fine to me.

jrbaudin reviewed (approved):
Looks good 👍

jrbaudin commented:
Sorry for the slowness of the review btw. It's been holidays and have been kidz focused.

ericcarraway commented:
@jrbaudin I've squashed this down to a single commit and rebased it on top of the latest updates to the main branch.  For posterity, here at are the original commits:

0dee843 add types GetSubscriberTags, GetSubscriberTagsInput
5a15ac0 add `.env.example` file for integration tests
382c51a add method `getSubscriberTags` to list the tags for a subscriber
